File: backend/src/shared_lib/entities/question.py
from typing import TypedDict

from pydantic import BaseModel

from entities.best_practice import BestPractice, BestPracticeID, RawBestPractice

QuestionID = str
PillarID = str


class Question(TypedDict):
    id: QuestionID
    primary_id: str
    label: str
    none: bool
    disabled: bool
    best_practices: dict[BestPracticeID, BestPractice]


class Pillar(TypedDict):
    id: PillarID
    primary_id: str
    label: str
    disabled: bool
    questions: dict[QuestionID, Question]


class RawQuestion(TypedDict):
    primary_id: str
    label: str
    best_practices: dict[str, RawBestPractice]


class RawPillar(TypedDict):
    primary_id: str
    label: str
    questions: dict[str, RawQuestion]


class QuestionDto(BaseModel):
    none: bool | None = None
    disabled: bool | None = None


class PillarDto(BaseModel):
    disabled: bool | None = None

File: libs/backend/models/src/lib/Question/Question.ts
import type { BestPractice, RawBestPractice } from '../BestPractice';

export interface Question {
  bestPractices: Record<string, BestPractice>;
  disabled: boolean;
  id: string;
  label: string;
  none: boolean;
  primaryId: string;
}

export interface QuestionBody {
  disabled?: boolean;
  none?: boolean;
}

export interface RawQuestion {
  primary_id: string;
  label: string;
  best_practices: Record<string, RawBestPractice>;
}
